images/agent/internal/controllers/drbd_primary/reconciler.go

```diff
@@ -0,0 +1,263 @@
+/*
+Copyright 2025 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package drbdprimary
+
+import (
+	"context"
+	"errors"
+	"os/exec"
+	"time"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/deckhouse/sds-replicated-volume/api/v1alpha3"
+	"github.com/deckhouse/sds-replicated-volume/images/agent/internal/env"
+	"github.com/deckhouse/sds-replicated-volume/images/agent/pkg/drbdadm"
+)
+
+type Reconciler struct {
+	cl     client.Client
+	log    logr.Logger
+	scheme *runtime.Scheme
+	cfg    env.Config
+}
+
+var _ reconcile.Reconciler = (*Reconciler)(nil)
+
+// NewReconciler is a small helper constructor that is primarily useful for tests.
+func NewReconciler(cl client.Client, log logr.Logger, scheme *runtime.Scheme, cfg env.Config) *Reconciler {
+	return &Reconciler{
+		cl:     cl,
+		log:    log,
+		scheme: scheme,
+		cfg:    cfg,
+	}
+}
+
+func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	log := r.log.WithName("Reconcile").WithValues("req", req)
+	log.Info("Reconciling started")
+	start := time.Now()
+	defer func() {
+		log.Info("Reconcile finished", "duration", time.Since(start).String())
+	}()
+
+	rvr := &v1alpha3.ReplicatedVolumeReplica{}
+	err := r.cl.Get(ctx, req.NamespacedName, rvr)
+	if err != nil {
+		log.Error(err, "getting ReplicatedVolumeReplica")
+		return reconcile.Result{}, err
+	}
+
+	if rvr.DeletionTimestamp != nil {
```

Comment:
Слава пишет
```suggestion
	if !rvr.DeletionTimestamp.IsZero() {
```
Он проверяет и `nil` и пустую структуру. На твой вкус